api/src/main/java/javax/enterprise/inject/spi/builder/AnnotatedTypeConfigurator.java

```diff
@@ -0,0 +1,432 @@
+
+    /**
+     * Reads the annotations from an existing java type. If overwrite is true
+     * then existing annotations will be overwritten
+     *
+     * @param type      the type to read from
+     * @param overwrite if true, the read annotation will replace any existing
+     *                  annotation
+     * @return self
+     */
+    <U extends T> AnnotatedTypeConfigurator<T> read(Class<U> type, boolean overwrite);
+
+    /**
+     * Override the declared type of a field
+     *
+     * @param field the field to override the type on
+     * @param type  the new type of the field
+     * @throws IllegalArgumentException if field or type is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> overrideFieldType(Field field, Type type);
+
+    /**
+     * Override the declared type of a field
+     *
+     * @param field the field to override the type on
+     * @param type  the new type of the field
+     * @throws IllegalArgumentException if field or type is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> overrideFieldType(AnnotatedField<? super T> field, Type type);
+
+    /**
+     * Override the declared type of a method parameter
+     *
+     * @param method   the method to override the parameter type on
+     * @param position the position of the parameter to override the type on
+     * @param type     the new type of the parameter
+     * @throws IllegalArgumentException if parameter or type is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> overrideMethodParameterType(Method method, int position, Type type);
+
+    /**
+     * Override the declared type of a constructor parameter
+     *
+     * @param constructor the constructor to override the parameter type on
+     * @param position    the position of the parameter to override the type on
+     * @param type        the new type of the parameter
+     * @throws IllegalArgumentException if parameter or type is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> overrideConstructorParameterType(Constructor<T> constructor, int position, Type type);
+
+    /**
+     * Override the declared type of a parameter.
+     *
+     * @param parameter the parameter to override the type on
+     * @param type      the new type of the parameter
+     * @throws IllegalArgumentException if parameter or type is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> overrideParameterType(AnnotatedParameter<? super T> parameter, Type type);
+
+    /**
+     * getter for the class represented by the AnnotatedType to configure
+     *
+     * @return Class represented by the AnnotatedType to configure
+     *
+     */
+    Class<T> getJavaClass();
+
+    /**
+     * set the class represented by the AnnotatedType to configure
+     *
+     * @param type the Class represented by the AnnotatedType
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> setType(Class<T> type);
```

Comment:
I think this method should be removed. Anyway, it's useless in its current form (you can only pass Class of the same type). Maybe we could make the method parameterized and allow to pass a subtype, i.e. `<U extends T>  AnnotatedTypeConfigurator<U> setType(Class<U> type)`. But this would for example break the contract of PAT imho.